ale_linters/java/javac.vim

```diff
@@ -0,0 +1,106 @@
+" Author: farenjihn <farenjihn@gmail.com>
+" Description: Lints java files using javac
+
+if exists('g:loaded_ale_linters_java_javac')
+    finish
+endif
+
+let g:loaded_ale_linters_java_javac = 1
+let g:ale_java_javac_classpath = ''
+
+let s:eclipse_classpath = ''
+let s:tmppath = '/tmp/java_ale/'
```

Comment:
We need to be compatible with windows as well.

I think instead of doing your file magic, we should rewrite the stdin-wrapper to use the same filename as the current filename.